docs/docs/guide/scale/tma-architecture.md

```diff
@@ -0,0 +1,93 @@
+
+### Reusability
+
+Reusing code across apps and other products like extensions is encouraged using frameworks or libraries. By building modules reusing them is pretty straightforward. We can build an iMessage extension, a Today Extension, or a watchOS application by just combining existing modules and adding _(when necessary)_ platform-specific UI layers.
+
+## Dependencies
+
+When a module depends on another module, it declares a dependency against its interface target. The benefit of this is two-fold. It prevents the implementation of a module to be coupled to the implementation of another module, and it speeds up clean builds because they only have to compile the implementation of our feature, and the interfaces of direct and transitive dependencies. This approach is inspired by SwiftRock's idea of [Reducing iOS Build Times by using Interface Modules](https://swiftrocks.com/reducing-ios-build-times-by-using-interface-targets).
+
+Depending on interfaces requires apps to build the graph of implementations at runtime, and dependency-inject it into the modules that need it. Although TMA is non-opinionated about how to do this, we recommend not using dependency-injection solutions that might add tooling indirection, make the logic harder to follow, or build on platform APIs that were not designed for this goal and therefore might make the solution brittle.
+
+## Product types
+
+When building a module, you can choose between **libraries and frameworks**, and **static and dynamic linking** for the targets. Without Tuist, making this decision is a bit more complex because you need to configure the dependency graph manually. However, thanks to Tuist Projects, this is no longer a problem. 
+
+We recommend using dynamic libraries or frameworks during development using [bundle accessors](/guide/project/synthesized-files#bundle-accessors) to decouple the bundle-accessing logic from the library or framework nature of the target. This is key for fast compilation times and to ensure [SwiftUI Previews](https://developer.apple.com/documentation/swiftui/previews-in-xcode) work reliably. And static libraries or frameworks for the release builds to ensure the app boots fast. You can leverage [dynamic configuration](/guide/project/dynamic-configuration#configuration-through-environment-variables) to change the product type at generation-time:
+
+```bash
+# You'll have to read the value of the variable from the manifest
+# and use it to change the linking type
+TUIST_PRODUCT_TYPE=static-library tuist generate
```

Comment:
Maybe worth adding a quick `Project.swift` snippet that shows how we use the environment variable there?